NMS-13273: Add docs about var-bind in notifications (#6955)
OpenNMS/opennms: docs/modules/operation/pages/deep-dive/notifications/concepts.adoc

If the switch is `off`, notices will never be sent to the target.
If the switch is `on`, notices will always be sent to the target.
If the switch is set to `auto`, the system deploys heuristics aimed at "doing the right thing."

> == Variable bindings

Reviewer: This is a perfect use case for this functionality: https://chat.opennms.com/opennms/pl/gnyui5sgwiddpfi1moymrrj8ry

Reviewer 2: Thanks for this feedback.  Honestly, I've never used either of those settings on notifications, so this doc update is welcomed.

Reviewer 2: According to the XSD, it is 1 varbind and many parameters
```
        <element maxOccurs="unbounded" minOccurs="0" ref="this:parameter"/>
        <element maxOccurs="1" minOccurs="0" ref="this:varbind"/>
```

If the switch is `off`, notices will never be sent to the target.
If the switch is `on`, notices will always be sent to the target.
If the switch is set to `auto`, the system deploys heuristics aimed at "doing the right thing."

== Variable bindings filters

Event parameters can additionally act as a filter method.
This can be helpful for events that don't have an association to a specific node and the common filters only match against interfaces.
By defining the `vbname` and `vbvalue`, the notification will only be sent with if the event parameter matches the given value.

Example 1:

> In this example, notifications on a SNMP Link Down event would be suppressed for trunk ports.

Reviewer 2: Wouldn't this be the opposite, only triggering events that do match `*Trunk*`?

Author: Yes. It is the positive match.

```suggestion
In this example, notifications would only be triggered on a SNMP Link Down events that contain the string `Trunk`.
```

.Example `notifications.xml` definition
[source, xml]
----
   <notification name="Link down notification" status="off">
      <uei>uei.opennms.org/translator/traps/SNMP_Link_Down</uei>
      <rule>(IPADDR IPLIKE *.*.*.*)</rule>
      <destinationPath>Email-Admin</destinationPath>
      <text-message>%descr%</text-message>
      <subject>Notice #%noticeid% - Link Down</subject>
      <numeric-message>111-%noticeid%</numeric-message>
      <varbind>
         <vbname>ifAlias</vbname> <1>
         <vbvalue>~^.*Trunk.*</vbvalue> <2>
      </varbind>
   </notification>
----
<1> Varbind or parameter name to filter
<2> Value to match for the specified varbind/parameter.
If the value starts with a `~`, it will be processed as a regular expression, otherwise it will be a string match.

Example 2:

> 

Reviewer 2: ```suggestion

In this example, Business Service Monitoring problem events would only trigger notifications if there is a parameter match with the specified business service name pattern.

```

If the switch is `off`, notices will never be sent to the target.
If the switch is `on`, notices will always be sent to the target.
If the switch is set to `auto`, the system deploys heuristics aimed at "doing the right thing."

== Variable bindings filters

> Event parameters can additionally act as a filter method.

Reviewer 2: ```suggestion
Event parameters can be used to filter which events trigger a notification.
```

> This can be helpful for events that don't have an association to a specific node and the common filters only match against interfaces.

Reviewer 2: ```suggestion
```

By defining the `vbname` and `vbvalue`, the notification will only be sent with if the event parameter matches the given value.

Example 1:

In this example, notifications would only be triggered on SNMP Link Down events that contain the string `Trunk`.

.Example `notifications.xml` definition
[source, xml]
----
   <notification name="Link down notification" status="off">
      <uei>uei.opennms.org/translator/traps/SNMP_Link_Down</uei>
      <rule>(IPADDR IPLIKE *.*.*.*)</rule>
      <destinationPath>Email-Admin</destinationPath>
      <text-message>%descr%</text-message>
      <subject>Notice #%noticeid% - Link Down</subject>
      <numeric-message>111-%noticeid%</numeric-message>
      <varbind>
         <vbname>ifAlias</vbname> <1>
         <vbvalue>~^.*Trunk.*</vbvalue> <2>
      </varbind>
   </notification>
----

> <1> Varbind or parameter name to filter

Author: ```suggestion
<1> Event parameter name to filter
```

> <2> Value to match for the specified varbind/parameter.

Author: ```suggestion
<2> Value to match for the specified event parameter.
```

If the value starts with a `~`, it will be processed as a regular expression, otherwise it will be a string match.

Example 2:

In this example, Business Service Monitoring problem events would only trigger notifications if there is a parameter match with the specified business service name pattern.

.Example `notifications.xml` definition
[source, xml]
----
   <notification name="BusinessServiceMonitor notification" status="off" writeable="yes">
      <uei>uei.opennms.org/bsm/serviceProblem</uei>
      <rule>(ipaddr != '0.0.0.0')</rule>
      <destinationPath>Email-Admin</destinationPath>
      <text-message>BSM Root Cause Analysis: %parm[rootCause]%</text-message>
      <subject>%parm[businessServiceName]% is impacted</subject>
      <numeric-message>111-%noticeid%</numeric-message>
      <varbind>
         <vbname>businessServiceName</vbname> <1>
         <vbvalue>~^MailCluster.*</vbvalue> <2>
      </varbind>
   </notification>
----

> <1> Varbind or parameter name to filter

Author: ```suggestion
<1> Event parameter name to filter
```

> <2> Value to match for the specified varbind/parameter.

Author: ```suggestion
<2> Value to match for the specified event parameter.
```
